### PR TITLE
Google Drive folders selection

### PR DESCRIPTION
**Pull request by lasryaric**
Here is what the folders selectors looks like:
Screen Shot 2023-06-12 at 16 32 51

We save the ids of the the folders "selected", even the ones that are selected because their children is selected. That's to be discussed but the react library I use claim to follow [some standard about managing all that](https://www.w3.org/WAI/ARIA/apg/#TreeView), so I decided to follow it.

Deployment:
There are Database changes on connectors side and connectors needs to be deployed before front.
I will see how to do all that in the deployment runbook.
I will setup the prod crendentials on Nango take care of the deployment.

### CODE REVIEW

**Review by spolu — DISMISSED** [review]
LGTM

Left a few questions for my own edition and a few suggestion, but feel free to merge once addressed :+1:

**Comment by lasryaric** [comment]
Deployment:
There are Database changes on connectors side and connectors needs to be deployed before front.
I will see how to do all that in the deployment runbook.
I will setup the prod crendentials on Nango take care of the deployment.

File: connectors/src/api/google_drive.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+export const googleDriveSetFoldersAPIHandler = withLogging(
+  _googleDriveSetFoldersAPIHandler
+);
+
+type GetFoldersRes =
+  | GoogleDriveSelectedFolderType[]
+  | ConnectorsAPIErrorResponse;
+
+const _googleDriveGetFoldersAPIHandler = async (
```
[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/api/google_drive.ts:100]: Thanks :+1: To be clear, what happens if a folder has multiple parents?

**Review by spolu — DISMISSED** [review]
LGTM

To update the DB you need to connect on a prod `core` machine, pull the code and run from `connectors/`
```
./admin/init_db.sh
```

File: connectors/src/api/google_drive.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lasryaric) [reply connectors/src/api/google_drive.ts:100]: Well, good question.
We explicitly take the first parent from the parents[] list.
I spent a good amount of time trying to put a file in multiple folders, and I can't figure out how to do it.
Also, it  looks like folders were never able to have multiple parents, only files, so in this context we are safe.

[the remark this answers is not in the document]
Comment (lasryaric) [reply connectors/src/api/google_drive.ts:100]: After searching one more time, I found this conversation : 
https://support.google.com/drive/thread/69017253/what-has-replaced-the-shift-z-function-on-google-drive-now-it-creates-short-cuts?hl=en

Which says:
```

it is no longer possible to have a file in multiple folders. This feature has been completely replaced by shortcuts.
 
If you use Google Drive's web interface this should not be an issue, however if you sync Google Drive to your computer using Bakcup & Sync, those shortcuts are downloaded simply as links to the online file.
```

[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/api/google_drive.ts:100]: Awesome

**Review by lasryaric — COMMENTED** [review]
@spolu I added 3 commits:
- The first 2 are fixing bugs in the folders selector UI (bug in the react-accessible-treeview and bug in our code. We were not re-fetching the folders when re-opening the Modal).
- the last one is for restarting the sync from scratch when the folders selection changes while we are syncing.

**Review by spolu — APPROVED** [review]
LGTM

A few unusued imports it seems?

**Pull request merged**